lib/ecto/migrator.ex

```diff
@@ -661,4 +665,49 @@ defmodule Ecto.Migrator do
 
   defp log(false, _msg), do: :ok
   defp log(level, msg),  do: Logger.log(level, msg)
+
+  defp ensure_repo_started(repo, pool_size) do
+    case repo.start_link(pool_size: pool_size) do
+      {:ok, _} ->
+        {:ok, :stop}
+
+      {:error, {:already_started, _pid}} ->
+        {:ok, :restart}
+
+      {:error, _} = error ->
+        error
+    end
+  end
+
+  defp ensure_migration_repo_started(repo, repo) do
+    {:ok, nil}
```

Comment:
Let's return `:noop` instead of `nil`, as it is clearer. :)

Reply:
👍  ✅  fixed